### PR TITLE
Assignment for chapter 9 unit testing

### PR DESCRIPTION
**Pull request by RitikaGarg-ITT**

### CODE REVIEW

File: divisorMatcher/src/services/DivisorMatcher.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  private numberService: INumberService;
+
+  constructor(numberService: INumberService) {
+    this.numberService = numberService;
+  }
+
+  findMatchingCount(n: number): number {
+    let count = 0;
+    for (let x = 1; x < n; x++) {
+      const y = n - x;
```
Comment (parakhmahaveer) [inline divisorMatcher/src/services/DivisorMatcher.ts]: can use meaningful naming

File: divisorMatcher/tests/DivisorMatcher.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  test("findMatchingCount for 1 should be 0", () => {
+    expect(matcher.findMatchingCount(1)).toBe(0);
+  });
+
+  test("findMatchingCount for 2 should be 1", () => {
+    expect(matcher.findMatchingCount(2)).toBe(1);
+  });
+
+  test("handle large input like 100 without crashing", () => {
+    expect(typeof matcher.findMatchingCount(100)).toBe("number");
```
Comment (parakhmahaveer) [inline divisorMatcher/tests/DivisorMatcher.test.ts:25]: Can add test case for 0 or negative number.

File: divisorMatcher/tests/DivisorMatcher.test.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    expect(matcher.findMatchingCount(10)).toBe(2);
+  });
+
+  test("findMatchingCount for 1 should be 0", () => {
+    expect(matcher.findMatchingCount(1)).toBe(0);
+  });
+
+  test("findMatchingCount for 2 should be 1", () => {
+    expect(matcher.findMatchingCount(2)).toBe(1);
+  });
```
Comment (RitikDave-Itt) [inline divisorMatcher/tests/DivisorMatcher.test.ts:22]: Can we add inline comments for logic clarity

File: divisorMatcher/src/services/DivisorMatcher.ts (hunk 1 above)
Comment (RitikaGarg-ITT) [reply divisorMatcher/src/services/DivisorMatcher.ts]: resolved
